phobos_umss.sh

```diff
@@ -0,0 +1,490 @@
+        STATUS="FAILURE"
+        save_log mv "STATUS=$STATUS($error_6)"
+    fi
+    save_log mv "END mv($*)"
+    return $error_6
+}
+
+# function to do a stat on a file $1 stored in phobos
+# Pour l'heure, on se contente d'extraire via $(phobos extent list -o media_name,address) le chemin du premier extent retourné dans le système de fichiers, et d'appeler $(stat) dessus pour appliquer les Regexps présentées dans le template des UMSS.
+stat () {
+    # <your command to retrieve stats on the file> $1
+    # e.g: output=$(/usr/local/bin/rfstat rfioServerFoo:$1)
+    save_log stat "BEGIN stat($*)"
+    op_ph=$(type -P phobos)
+    json_output_script="sudo $op_ph object list ${1} -t -o user_md -f human"
+    json_output=$($json_output_script)
+    error_7=$?
+    save_log stat "$json_output_script return with status=$json_output($error_7)"
+    if [ $error_7 == 0 ]; then
+        keys_order=("device" "inode" "mode" "nlink" "uid" "gid" "devid" "size" "blksize" "blkcnt" "atime" "mtime" "ctime")
+        irods_output=$(echo "$json_output" | jq -r --argjson keys "$(printf '%s\n' "${keys_order[@]}" | jq -R . | jq -s .)" '
+  [
+    . as $in | $keys[] | $in[.]
+  ] | join(":")
+')
+        error_7=$?
+	save_log stat "function ends with status=($error_7), exporting string:\n${irods_output}"
+        if [ $error_7 == 0 ]; then
+            echo "${irods_output}"
+        else
+            echo "0:0:0:0:0:0:0:0:0:0:0:0:0"
+        fi
+    fi
+    
+    if [ $error_7 != 0 ]; then
+        STATUS="FAILURE"
+        save_log stat "STATUS=$STATUS($error_7)"
+    fi
+    save_log stat "END stat($*)"
+    return $error_7
+}
+
+#######################
+# Secondary functions #
+#######################
+
+# Outputs a log line in stdout, or in $LOGFILE if any.
+save_log() {
+    if [ $LEVEL -eq $DEBUG ]; then
+        timestamp=$(date +"%Y:%m:%d-%T.%N")
+        # Messages building
+        if [ "$2" ]; then
+            message_colour="${Black}${ID} ${Green}[${timestamp}] ${Red}${1}(${Blue}${2}${Red})${Color_Off}"
+            message_black="${ID} [${timestamp}] ${1}(${2})"
+        else
+            message_colour="${Black}${ID} ${Red}${1}${Color_Off}"
+            message_black="${ID} ${1}"
+        fi
+
+        # Message(s) writing on file(s)
+        if [ -n "${LOGFILE_FANCY+x}" ]; then
+            if [ "$LOGFILE_FANCY" != "" ]; then
+                echo -e "$message_colour" >> "$LOGFILE_FANCY"
+            fi
+        fi
+
+        if [ -n "${LOGFILE_SOBRE+x}" ]; then
+            if [ "$LOGFILE_SOBRE" != "" ]; then
+                echo -e "$message_black" >> "$LOGFILE_SOBRE"
+            fi
+        fi
+
+        # Message(s) displaying on screen
+        if [ $DISPLAY == 1 ]; then
+            if [ $DISPLAY_COLOUR == 1 ]; then
+                echo -e "$message_colour"
+            fi
+            if [ $DISPLAY_BLACK == 1 ]; then
+                echo -e "$message_black"
+            fi
+        fi
+    fi
+    return 0
+}
```

Comment:
minor: you can add a way to parameter the logs into a file, or a system log ?

Reply:
Excuse me, what do you mean ? Introducing multiple log levels ?